typescript/function.go

```diff
@@ -0,0 +1,57 @@
+package typescript
+
+type Function struct {
```

Comment:
want to add support for arrow functions? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

Reply:
good call. ill look at it

Reply:
ok, added support for arrow, give it another look. 

This is currently only supporting the most general form:
```
(param1, paramN) => {
  statements
}
```